skbio/io/fastq.py

```diff
@@ -480,21 +480,30 @@ def _parse_sequence_data(fh):
 def _parse_quality_scores(fh, seq_len, variant, phred_offset):
     phred_scores = []
     qual_len = 0
-    for chunk in _line_generator(fh):
-        if chunk.startswith('@') and qual_len == seq_len:
-            return phred_scores, chunk
+    prev_was_blank = False
+    for chunk in fh:
+        chunk = chunk.strip()
+        if chunk:
+            if chunk.startswith('@') and qual_len == seq_len:
+                return phred_scores, chunk
+            else:
+                if prev_was_blank:
+                    raise FASTQFormatError("Found blank line within FASTQ "
```

Comment:
Same comment as above, copied here:

Can you update this error message to indicate blank or whitespace-only (similar to FASTA error message)? It only mentioned blank before because `rstrip('\n')` was being used, but now all leading/trailing whitespace is stripped.

It'd be nice if these two error paths could become one again... but I can't think of a good way to do this (in `_line_generator` or elsewhere) since you have to error out *after* grabbing the next line.